buildSrc/src/main/java/com/uber/okbuck/core/dependency/DependencyUtils.java

```diff
@@ -32,37 +39,34 @@ public static Configuration useful(Project project, String configuration) {
 
   @Nullable
   public static Configuration useful(@Nullable Configuration configuration) {
-    return configuration != null ? (configuration.isCanBeResolved() ? configuration : null) : null;
-  }
-
-  public static File createCacheDir(
-      Project project, String cacheDirPath, @Nullable String buckFile) {
-    File cacheDir = project.getRootProject().file(cacheDirPath);
-    cacheDir.mkdirs();
-
-    if (buckFile != null) {
-      FileUtil.copyResourceToProject(buckFile, new File(cacheDir, "BUCK"));
+    if (configuration != null && configuration.isCanBeResolved()) {
+      return configuration;
     }
-
-    return cacheDir;
+    return null;
   }
 
-  public static File createCacheDir(Project project, String cacheDirPath) {
-    return DependencyUtils.createCacheDir(project, cacheDirPath, null);
-  }
-
-  public static boolean isWhiteListed(final File depFile) {
+  public static boolean isWhiteListed(final File dependencyFile) {
     return WHITELIST_LOCAL_PATTERNS
         .stream()
-        .anyMatch(pattern -> depFile.getPath().contains(pattern));
+        .anyMatch(pattern -> dependencyFile.getPath().contains(pattern));
   }
 
   public static boolean isConsumable(File file) {
+    // Skip artifact files which are coming from the transformed folder.
+    // transforms-1 contains the contents of the resolved aar/jar and
+    // hence should not be consumed.
+    if (file.getAbsolutePath().contains("transforms-1/files-1")) {
+      return false;
+    }
```

Comment:
@kageiit need this to prevent jars from extracted aar's which are being added multiple times for the same artifact with different dependency files.

Reply:
SG. Documentation can read "Gradle artifact api transform aars/jars" instead of "resolved aars/jars" to be more clear. We can fix up documentation in a subsequent pass